pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -395,6 +400,16 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 			r.status.SetDegraded("Failed to get eck trial license", err.Error())
 			return reconcile.Result{}, err
 		}
+
+		if elasticLicenseType, err = utils.GetElasticLicenseType(ctx, r.client, reqLogger); err != nil {
+			// If ECKLicenseConfigMapName is not found, it means ECK operator is not running yet, log the information and proceed
```

Comment:
I'm just commenting for the sake of it, but maybe it would make sense for separate controllers for the ECK operator and Elasticsearch?

Please don't consider this for this PR, I'm just putting this thought out inter the ether.